app/models/miq_expression.rb

```diff
@@ -1128,11 +1128,33 @@ def self.model_details(model, opts = {:typ => "all", :include_model => true, :in
       result = get_column_details(relats[:columns], model, model, opts).sort! { |a, b| a.to_s <=> b.to_s }
       result.concat(tag_details(model, model, opts)) if opts[:include_tags] == true
     end
-    result.concat(_model_details(relats, opts).sort! { |a, b| a.to_s <=> b.to_s })
+
+    model_details = _model_details(relats, opts)
+
+    custom_details = _custom_details_for(model)
+    model_details.concat(custom_details) unless custom_details.empty?
+
+    result.concat(model_details.sort! { |a, b| a.to_s <=> b.to_s })
```

Comment:
not sure how many results are here.
But you may get a boost using model_details.sort_by!(&:to_s)